Add issue forms

eddumelendez opened the pull request:
Introducing issue forms for: bug report, enhacements and features.
In case of help or question there is a link to the slack channel.

eddumelendez commented:
You can see a preview of those forms [here](https://github.com/eddumelendez/tc-issue-form/issues/new/choose)

kiview commented:
Wow, this is super cool, thanks for looking into this 🙌 

I have one question regarding the intent of this selection field:
![image](https://user-images.githubusercontent.com/5088104/173569570-b7b50f08-da27-4952-bbf6-2c9e85dc86b7.png)

**Yes** means: _I used the old version, updated to the new version, and the issue persists._
**No** means: _I used the old version, updated to the new version, and the issue is solved._

Is it correct? Does this mainly act for nudging users into trying out with the latest version?

kiview commented on .github/ISSUE_TEMPLATE/enhancement.yaml:
```diff
@@ -0,0 +1,56 @@
+    id: module
+    attributes:
+      label: Module
+      description: Which testcontainers module applies the enhancement proposal?
```
```suggestion
      description: For which Testcontainers module does the enhancement proposal apply?
```

kiview commented on .github/ISSUE_TEMPLATE/enhancement.yaml:
```diff
@@ -0,0 +1,56 @@
+    id: proposal
+    attributes:
+      label: Proposal
+      description: Also tell us, what did you expect to happen?
```
```suggestion
      description: What should be improved? What are the limitations of the current implications that would be solved by the proposal?
```

kiview commented on .github/ISSUE_TEMPLATE/feature.yaml:
```diff
@@ -0,0 +1,85 @@
+        - Solr
+        - ToxiProxy
+        - Trino
+        - Vault
```
```suggestion
        - Vault
        - New Module
```

kiview commented on .github/ISSUE_TEMPLATE/feature.yaml:
```diff
@@ -0,0 +1,85 @@
+name: Feature
+description: File a feature
```
```suggestion
description: Suggest a new feature
```

kiview commented on .github/ISSUE_TEMPLATE/feature.yaml:
```diff
@@ -0,0 +1,85 @@
+    id: solution
+    attributes:
+      label: Solution
+      description: What's the proposal solution for this feature?
```
```suggestion
      description: What's the proposed solution for this feature?
```

kiview commented on .github/ISSUE_TEMPLATE/feature.yaml:
```diff
@@ -0,0 +1,85 @@
+    id: problem
+    attributes:
+      label: Problem
+      description: Is your feature related to a problem? Please describe
```
```suggestion
      description: Is this feature related to a problem? Please describe it.
```

kiview commented on .github/ISSUE_TEMPLATE/feature.yaml:
```diff
@@ -0,0 +1,85 @@
+    id: benefit
+    attributes:
+      label: Benefit
+      description: What's the benefit by implementing it in the project?
```
```suggestion
      description: What's the benefit of addng this feature to the project?
```

kiview commented on .github/ISSUE_TEMPLATE/feature.yaml:
```diff
@@ -0,0 +1,85 @@
+    id: alternatives
+    attributes:
+      label: Benefit
+      description: Are there other alternatives? Please describe
```
```suggestion
      description: Are there other alternatives? Please describe them.
```

kiview commented on .github/ISSUE_TEMPLATE/feature.yaml:
```diff
@@ -0,0 +1,85 @@
+    id: contribute
+    attributes:
+      label: Would you like to help contributing this feature?
+      description: Which testcontainers module are you using?
```
```suggestion
```

kiview reviewed (approved):
Some suggestions for different wording, accept as you see fit. Really excited to see the results of using these forms 🤗

eddumelendez commented:
> **Yes** means: _I used the old version, updated to the new version, and the issue persists._ 
**No** means: _I used the old version, updated to the new version, and the issue is solved._
> 
> Is it correct? Does this mainly act for nudging users into trying out with the latest version?

Yes, that's the meaning in the reply. Although, the main act is not updating to the latest but identify if the issue was already fixed in a later version (not necessarily latest one)
